// react-login-page/cjs/Render.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Render = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _react = require("react");
var _store = require("./store");
var _jsxRuntime = require("react/jsx-runtime");
var Render = exports.Render = function Render(_ref) {
  var children = _ref.children;
  var _useStore = (0, _store.useStore)(),
    _useStore$fields = _useStore.fields,
    fields = _useStore$fields === void 0 ? {} : _useStore$fields,
    _useStore$buttons = _useStore.buttons,
    buttons = _useStore$buttons === void 0 ? {} : _useStore$buttons,
    _useStore$extra = _useStore.extra,
    extra = _useStore$extra === void 0 ? {} : _useStore$extra,
    _useStore$$$index = _useStore.$$index,
    $$index = _useStore$$$index === void 0 ? {} : _useStore$$$index,
    _useStore$blocks = _useStore.blocks,
    blocks = _useStore$blocks === void 0 ? {} : _useStore$blocks,
    data = _useStore.data;
  var childs = typeof children === 'function' ? [] : _react.Children.toArray(children);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)(_react.Fragment, {
    children: [typeof children === 'function' && ! /*#__PURE__*/(0, _react.isValidElement)(children) && children({
      fields: fields,
      buttons: buttons,
      blocks: blocks,
      extra: extra,
      $$index: $$index
    }, (0, _objectSpread2["default"])({}, data)), typeof children !== 'function' && childs.map(function (child, key) {
      if (! /*#__PURE__*/(0, _react.isValidElement)(child)) return null;
      return /*#__PURE__*/(0, _react.cloneElement)(child, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, child.props), {}, {
        key: key
      }));
    })]
  });
};
Render.displayName = 'Login.Render';
