// react-login-page/cjs/store.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.initialState = exports.Provider = exports.Context = void 0;
exports.reducer = reducer;
exports.useStore = void 0;
var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _react = _interopRequireWildcard(require("react"));
var _jsxRuntime = require("react/jsx-runtime");
var initialState = exports.initialState = {
  index: {},
  fields: {},
  buttons: {},
  blocks: {},
  extra: {},
  data: {
    fields: [],
    buttons: [],
    blocks: []
  }
};
var Context = exports.Context = /*#__PURE__*/(0, _react.createContext)(initialState);
Context.displayName = 'Login.Context';
function reducer(state, action) {
  var result = (0, _objectSpread2["default"])((0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state), action), {}, {
    $$index: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state.$$index), action.$$index),
    fields: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state.fields), action.fields),
    buttons: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state.buttons), action.buttons),
    blocks: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state.blocks), action.blocks),
    extra: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state.extra), action.extra)
  });
  var fieldsArray = Object.keys(result.fields).map(function (key) {
    var _result$fields$key;
    return {
      name: key,
      index: ((_result$fields$key = result.fields[key]) === null || _result$fields$key === void 0 || (_result$fields$key = _result$fields$key.props) === null || _result$fields$key === void 0 ? void 0 : _result$fields$key.index) || (result.$$index || {})[key] || 0,
      children: result.fields[key]
    };
  });
  var buttonsArray = Object.keys(result.buttons).map(function (key) {
    var _result$buttons$key;
    return {
      name: key,
      index: ((_result$buttons$key = result.buttons[key]) === null || _result$buttons$key === void 0 || (_result$buttons$key = _result$buttons$key.props) === null || _result$buttons$key === void 0 ? void 0 : _result$buttons$key.index) || (result.$$index || {})[key] || 0,
      children: result.buttons[key]
    };
  });
  var blocksArray = Object.keys(result.blocks).map(function (key) {
    var _result$blocks$key;
    return {
      name: key,
      index: ((_result$blocks$key = result.blocks[key]) === null || _result$blocks$key === void 0 || (_result$blocks$key = _result$blocks$key.props) === null || _result$blocks$key === void 0 ? void 0 : _result$blocks$key.index) || (result.$$index || {})[key] || 0,
      children: result.blocks[key]
    };
  });
  return (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, result), {}, {
    data: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, result.data), {}, {
      fields: fieldsArray,
      buttons: buttonsArray,
      blocks: blocksArray
    })
  });
}
var useStore = exports.useStore = function useStore() {
  return (0, _react.useContext)(Context);
};
var Provider = exports.Provider = function Provider(_ref) {
  var children = _ref.children,
    render = _ref.render;
  var _useReducer = (0, _react.useReducer)(reducer, initialState),
    _useReducer2 = (0, _slicedToArray2["default"])(_useReducer, 2),
    state = _useReducer2[0],
    dispatch = _useReducer2[1];
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(Context.Provider, {
    value: (0, _objectSpread2["default"])((0, _objectSpread2["default"])({}, state), {}, {
      dispatch: dispatch
    }),
    children: children
  });
};
Provider.displayName = 'Login.Provider';
